packages/flutter/lib/src/material/button_style.dart

```diff
@@ -159,18 +160,29 @@ class ButtonStyle with Diagnosticable {
   ///
   /// The size of the rectangle the button lies within may be larger
   /// per [tapTargetSize].
+  ///
+  /// This value must be less than or equal to [maximumSize].
```

Comment:
Where is this enforced?

Reply:
In the _ButtonStyleStat.build() expression that builds a BoxConstraints object:
```dart
    BoxConstraints effectiveConstraints = resolvedVisualDensity.effectiveConstraints(
      BoxConstraints(
        minWidth: resolvedMinimumSize!.width,
        minHeight: resolvedMinimumSize.height,
        maxWidth: resolvedMaximumSize!.width,
        maxHeight: resolvedMaximumSize.height,
      ),
    );
```

If you pass invalid minimum/maximumSize values you get a pretty obvious error from the BoxConstraints constructor.